rollodex/src/App.js

```diff
@@ -1,26 +1,98 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, {Component} from 'react';
+// import logo from './logo.svg';
 import './App.css';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isHidden: true, contacts: [],
+    }
+  }
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData() {
+    fetch ('https://randomuser.me/api?results=25')
+    .then(response => response.json())
+    .then(parsedJSON => parsedJSON.results.map(contact => ({
+      name: `${contact.name.first} ${contact.name.last}`,
+      location: `${contact.location.city}, ${contact.location.state}, ${contact.location.country}`,
+      email: `${contact.email}`,
+      cell: `${contact.cell}`,
+      username: `${contact.login.username}`,
+      mediumpic: `${contact.picture.medium}`,
+      smallpic: `${contact.picture.thumbnail}`
+    })))
+    .then(contacts => this.setState({
+      contacts,
+      isHidden: true
+    }))
+    .catch(error => console.log("parsing failed", error))
+  }
+
+  handleClick = () => {
+    let status = this.state.isHidden === true ? false : true
+    this.setState({
+      isHidden: status
+    })
+  }
```

Comment:
Just realized my mistake above... I see you're hiding/showing the whole section.  It's supposed to hide/show just that one user.  With your setup (that I commented on above) you could do _something_ like this:

```
toggleDetails = (toggleIndex) => {   // could also use `id`
  let newContacts = this.state.contacts.map( (contact,index) => {
    return { ...contact, isShown: (index === toggleIndex) ? !contact.isShown : contact.isShown) };
  });
  this.setState({ contacts: contacts });
}
...
<div>
  { contacts.map( (contact,index) => (
    <button onClick={() => this.toggleDetails(index)}>Toggle Details</button>
  )}
</div>
```